roxie/ccd/ccdfile.cpp

```diff
@@ -2882,7 +2895,18 @@ class CRoxieWriteHandler : implements IRoxieWriteHandler, public CInterface
             if (localCluster)
                 throw MakeStringException(0, "Cluster %s occupies node already specified while writing file %s",
                         cluster, dFile->queryLogicalName());
-            localCluster.setown(group.getClear());
+            SocketEndpointArray eps;
+            SocketEndpoint me(0, getNodeAddress(myNodeIndex));
+            eps.append(me);
+            if (getNumNodes() > 1)
+            {
+                unsigned buddy = myNodeIndex+1;
+                if (buddy >= getNumNodes())
+                    buddy = 0;
+                SocketEndpoint buddyNode(0, getNodeAddress(buddy));
+                eps.append(buddyNode);
+            }
+            localCluster.setown(createIGroup(eps));
             localClusterName.set(cluster);
```

Comment:
I think this needs to be a valid match/lookup to the ip's in the group.
So could use : eps.getText(<StringBuffer>);